sdk/tests/platform/test_az_mqtt5_policy.c

```diff
@@ -104,79 +106,82 @@ static az_result test_inbound_hfsm_root(az_event_policy* me, az_event event)
-      assert_int_equal(
-          az_mqtt5_property_bag_stringpair_find(
-              test_recv_data->properties,
-              AZ_MQTT5_PROPERTY_TYPE_USER_PROPERTY,
-              AZ_SPAN_FROM_STR(TEST_MQTT_PROPERTY_STRING_PAIR_KEY1),
-              &test_mqtt5_property_string_pair1),
-          AZ_OK);
-      assert_true(az_span_is_content_equal(
-          az_mqtt5_property_stringpair_key_get(&test_mqtt5_property_string_pair1),
-          AZ_SPAN_FROM_STR(TEST_MQTT_PROPERTY_STRING_PAIR_KEY1)));
-      assert_true(az_span_is_content_equal(
-          az_mqtt5_property_stringpair_value_get(&test_mqtt5_property_string_pair1),
-          AZ_SPAN_FROM_STR(TEST_MQTT_PROPERTY_STRING_PAIR_VALUE1)));
-      az_mqtt5_property_stringpair_free(&test_mqtt5_property_string_pair1);
-
-      az_mqtt5_property_stringpair test_mqtt5_property_string_pair2;
-      assert_int_equal(
-          az_mqtt5_property_bag_stringpair_find(
-              test_recv_data->properties,
-              AZ_MQTT5_PROPERTY_TYPE_USER_PROPERTY,
-              AZ_SPAN_FROM_STR(TEST_MQTT_PROPERTY_STRING_PAIR_KEY2),
-              &test_mqtt5_property_string_pair2),
-          AZ_OK);
-      assert_true(az_span_is_content_equal(
-          az_mqtt5_property_stringpair_key_get(&test_mqtt5_property_string_pair2),
-          AZ_SPAN_FROM_STR(TEST_MQTT_PROPERTY_STRING_PAIR_KEY2)));
-      assert_true(az_span_is_content_equal(
-          az_mqtt5_property_stringpair_value_get(&test_mqtt5_property_string_pair2),
-          AZ_SPAN_FROM_STR(TEST_MQTT_PROPERTY_STRING_PAIR_VALUE2)));
-      az_mqtt5_property_stringpair_free(&test_mqtt5_property_string_pair2);
-
-      uint32_t test_mqtt5_property_int_32;
-      assert_int_equal(
-          az_mqtt5_property_bag_int_read(
-              test_recv_data->properties,
-              AZ_MQTT5_PROPERTY_TYPE_MESSAGE_EXPIRY_INTERVAL,
-              &test_mqtt5_property_int_32),
-          AZ_OK);
-      assert_int_equal(test_mqtt5_property_int_32, TEST_MQTT_PROPERTY_MESSAGE_EXPIRY_INTERVAL);
-
-      uint8_t test_mqtt5_property_int_8;
-      assert_int_equal(
-          az_mqtt5_property_bag_byte_read(
-              test_recv_data->properties,
-              AZ_MQTT5_PROPERTY_TYPE_PAYLOAD_FORMAT_INDICATOR,
-              &test_mqtt5_property_int_8),
-          AZ_OK);
-      assert_int_equal(test_mqtt5_property_int_8, TEST_MQTT_PROPERTY_PAYLOAD_FORMAT_INDICATOR);
-
-      az_mqtt5_property_binarydata test_mqtt5_property_binary_data;
-      assert_int_equal(
-          az_mqtt5_property_bag_binarydata_read(
-              test_recv_data->properties,
-              AZ_MQTT5_PROPERTY_TYPE_CORRELATION_DATA,
-              &test_mqtt5_property_binary_data),
-          AZ_OK);
-      assert_true(az_span_is_content_equal(
-          test_mqtt5_property_binary_data.bindata,
-          AZ_SPAN_FROM_BUFFER(TEST_MQTT_PROPERTY_CORRELATION_DATA)));
-      az_mqtt5_property_binarydata_free(&test_mqtt5_property_binary_data);
+      if (expect_msg_properties)
```

Comment:
If we're not exposing a check to see if the property bag is empty, then these functions should still be able to run and return a failure if there aren't any properties - I would propose the test to have an else on this if that looks like this:
```suggestion
if (!expect_msg_properties)
      az_mqtt5_property_string test_mqtt5_property_string;
      assert_int_equal(
          az_mqtt5_property_bag_string_read(
              test_recv_data->properties,
              AZ_MQTT5_PROPERTY_TYPE_CONTENT_TYPE,
              &test_mqtt5_property_string),
          AZ_ERROR_ITEM_NOT_FOUND);
      // not sure what this should be checked against here
      assert_true(az_span_is_content_equal(
          az_mqtt5_property_string_get(&test_mqtt5_property_string),
          AZ_SPAN_EMPTY));
      az_mqtt5_property_string_free(&test_mqtt5_property_string);
```